staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,174 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	authenticationv1alpha1client "k8s.io/client-go/kubernetes/typed/authentication/v1alpha1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/templates"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/yaml"
+)
+
+// WhoAmIOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type WhoAmIOptions struct {
+	AuthClient authenticationv1alpha1client.AuthenticationV1alpha1Interface
+
+	TablePrinter printers.ResourcePrinter
+	OutputFormat *string
+
+	genericclioptions.IOStreams
+	PrintFlags *genericclioptions.PrintFlags
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl auth who-am-i
+		
+		# Get your subject attributes in JSON format.
+		kubectl auth who-am-i -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth who-am-i' sub command
+func NewCmdWhoAmI(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &WhoAmIOptions{
+		OutputFormat: pointer.String(""),
+		IOStreams:    streams,
+		PrintFlags:   genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "who-am-i",
+		DisableFlagsInUseLine: true,
+		Short:                 "Check self subject attributes.",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.RunAttributesGet())
+		},
+	}
+
+	allowedFormats := []string{"yaml", "json"}
+	cmd.Flags().StringVarP(o.OutputFormat, "output", "o", *o.OutputFormat, fmt.Sprintf(`Output format. One of: (%s).`, strings.Join(allowedFormats, ", ")))
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *WhoAmIOptions) Complete(f cmdutil.Factory, args []string) error {
+	c, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+
+	o.AuthClient = c.AuthenticationV1alpha1()
+
+	o.TablePrinter = printers.NewTablePrinter(printers.PrintOptions{AllowMissingKeys: true})
+	return nil
+}
+
+// RunAttributesGet prints all user attributes.
+func (o *WhoAmIOptions) RunAttributesGet() error {
+	sar := &authenticationv1alpha1.SelfSubjectAttributesReview{}
+	response, err := o.AuthClient.SelfSubjectAttributesReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if o.OutputFormat == nil {
```

Comment:
@soltysh is this the current style in `kubectl`?  I thought this would end up in a validate method.

Reply:
No it is not, this is very wrong, all the printing should be re-using built-in printers, as mentioned in other comments. Whereas all the validation should be part of `Validate` method for ease of re-use by other actors.

Reply:
This one is also fixed.